sources/org.osbuild.curl

```diff
@@ -86,6 +87,59 @@ SCHEMA = """
 """
 
 
+class Subscriptions:
+    def __init__(self, sections):
+        self.sections = sections
+
+    @classmethod
+    def from_host_system(cls):
+        with open('/etc/yum.repos.d/redhat.repo', 'r') as fp:
+            return cls._parse_repo_file(fp)
+
+    @staticmethod
+    def _process_baseurl(input_url):
+        # First escape meta characters: . ^ $ * + ? { } [ ] \ | ( )
+        # Except for $basearch, which will be replaced later.
+        for character in ['\\', '.', '^', '*', '+', '?', '{', '}', '[', ']', '|', '(',')']:
+            input_url = input_url.replace(character, f"\\{character}")
+
+        # Now replace variables with regexes (see man 5 yum.conf for the list)
+        for variable in ['$releasever', '$arch', '$basearch', '$uuid']:
+            input_url = input_url.replace(variable, '.*')
+
+        if input_url.endswith('/'):
+            input_url = input_url + '.*'
+        else:
+            input_url = input_url + '/.*'
```

Comment:
This should not be needed if you use `re.match()` in `get_secrets()` as it matches the string from the beginning, but not to the end of it.

Reply:
Perfect, combined with `re.escape` it makes the function much shorter :-)